lib/helpers/calculator_helpers.js

```diff
@@ -116,12 +116,14 @@ module.exports = class CalculatorHelpers {
         }
         if ('observation_values' in populationResults) {
           // DENOM observation will be the first of 2 observations
+          populationResultsHandled.observation_values[1] = populationResultsHandled.observation_values[0];
           populationResultsHandled.observation_values[0] = 0;
         }
       }
       if (populationResults.DENEX != null && !this.isValueZero('DENEX', populationResults) && populationResults.DENEX >= populationResults.DENOM) {
         if ('observation_values' in populationResults) {
           // DENOM observation will be the first of 2 observations
+          populationResultsHandled.observation_values[1] = populationResultsHandled.observation_values[0];
```

Comment:
I do not believe you want to upshift the observation here.

In this case (Where the patient calculates into the DENEX and DENOM) there will value for the Denominator Observation in the 0 index of the observation_values array.  If the patient happens to be in the Numerator, the Numerator Observation will already be in the 1 index of the observation_value array.

If this code change is kept, then the Denominator Observation value will incorrectly override the Numerator Observation in scenarios when a patient meets the DENOM/DENEX and NUMER criteria.

Here are some screen captures showing the results from a sample Ratio measure I've attached (as CQL)
[SampleRatio.cql.zip](https://github.com/projecttacoma/cqm-execution/files/14270159/SampleRatio.cql.zip)

**This is the original calculation.  The NUMER Observation is missing when a patient is in the NUMER and not in the DENOM.**

![Original - With Bug](https://github.com/projecttacoma/cqm-execution/assets/8173551/fe741ecb-e412-4d0e-9629-b814c159a102)

**This is the calculation for this PR as written.  The DENOM Observation (of 1) is overriding the NUMER Observation (of 2)**

![PR - As written](https://github.com/projecttacoma/cqm-execution/assets/8173551/865eb126-38a1-4c0a-839b-4067c52b4c7a)

**This is the calculation without this line of code (Line 126 above).  The NUMER Observation of 2 is not being overridden.**

![PR - Without line 126 ](https://github.com/projecttacoma/cqm-execution/assets/8173551/ee0f48e0-42c1-4e11-ade6-cd80b442e523)

Reply:
@dczulada I was able to reproduce this scenario, where the same episode is part of IPP, DENOM, DENEX and NUM
Working on addressing this scenario right now, I will have a commit shortly

Reply:
Thank you so much, I reverted it and added a new test case to make sure.

Reply:
looks like looking good now. works in Bonnie.